src/rfc3986/validators.py

```diff
@@ -50,9 +53,9 @@ class Validator:
 
     def __init__(self):
         """Initialize our default validations."""
-        self.allowed_schemes = set()
-        self.allowed_hosts = set()
-        self.allowed_ports = set()
+        self.allowed_schemes: set[str] = set()
+        self.allowed_hosts: set[str] = set()
+        self.allowed_ports: set[str] = set()
```

Comment:
When did `set[type]` start work? I thought this needed to be `t.Set[str]`

Reply:
As far as I can tell, it's because it doesn't need to be evaluated. For comparison, at global scope, a variable annotation needs to be evaluated and stored in the module's `__annotations__` dict, and a function's annotations need at any scope need to be evaluated and stored in the function's `__annotations__` dict, even if it isn't immediately.

A lone variable annotation that isn't at global scope though? That's not evaluated or stored anywhere, as far as I can tell, so one can use the generic versions of the builtin types pre-3.9 without consequence. That's my understanding, at least. Not sure if it's explicitly documented somewhere.